File: julia/leap/leap.jl
"""
    is_leap_year(year)

Return `true` if `year` is a leap year in the gregorian calendar.
    on every year that is evenly divisible by 4
    except every year that is evenly divisible by 100
      unless the year is also evenly divisible by 400
"""
function is_leap_year(year)
    if year % 4 == 0 && year % 100 != 0
		return true 
	elseif year % 4 == 0 && year % 400 == 0
        return true 
	else
		return false
    end

end

File: csharp/leap/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        if (year % 4 == 0){
            bool isDivisibleby100 = year % 100 == 0;
            bool isDivisibleby400 = year % 400 == 0;
            if(isDivisibleby100){
                if(isDivisibleby400){
                    return true;
                }
                else{
                    return false;
                }
            }
            else{
                return true;
            }
        }
        else{
            return false;
        }
    }
}